packages/block-editor/src/components/rich-text/native/get-format-colors.native.js

```diff
@@ -5,50 +5,43 @@ import { getColorObjectByAttributeValues } from '../../../components/colors';
 
 const FORMAT_TYPE = 'core/text-color';
 const REGEX_TO_MATCH = /^has-(.*)-color$/;
-const TAGS_TO_SEARCH = /\<mark/;
 
-export function getFormatColors( value, formats, colors ) {
-	if ( value?.search( TAGS_TO_SEARCH ) !== -1 ) {
```

Comment:
I presume this condition was added to improve performance, but as shared in the PR's description, looping the `formats` array doesn't look enough expensive to need it. Plus, a regex search is probably adding some unneeded overhead.